src/pages/settings/Profile/Contacts/ContactMethodDetailsPage.js

```diff
@@ -142,6 +142,12 @@ class ContactMethodDetailsPage extends Component {
 
     render() {
         const contactMethod = this.getContactMethod();
+
+        // replacing spaces with "hard spaces" to prevent breaking the number
+        const formattedContactMethod = Str.isSMSLogin(contactMethod)
+            ? this.props.formatPhoneNumber(Str.removeSMSDomain(contactMethod)).replace(/ /g, '\u00A0')
```

Comment:
Why do we do this here rather than in the `formatPhoneNumber` method itself?

Reply:
I thought that because we don't need everywhere (just in 3 cases) I won't be replacing spaces in the util function. It is also a bit unpleasant for testing this functions and any other utils using them - you have to remember about this while writing test cases - it looks a bit confusing when test don't pass
![Screenshot 2023-04-11 at 17 41 23](https://user-images.githubusercontent.com/28020445/231216271-eb67aad0-dcea-4838-8ce4-860fb04696cf.png)

Reply:
Ok, makes sense.